feature/experimental/gribi/ate_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,668 @@
+	}("gRIBi Add")
+
+	go func(msg string) {
+		switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+		if err != nil {
+			t.Logf("Failed to perform control processor switchover with unexpected err: %v", err)
+		}
+		t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+	}("Master Switchover")
+
+	t.Log("Concurrent switchover and route addition is completed, validate switchoverStatus now.")
+
+	primaryAfterSwitch := validateSwitchoverStatus(t, dut, secondaryBeforeSwitch)
+
+	validateSwitchoverTelemetry(t, dut, primaryAfterSwitch)
+
+	// Following reconnection of the gRIBI client to a new master supervisor,
+	// validate if partially deleted entries of ipBlock1  are not present in the FIB
+	// using a get RPC.
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	// Check vars for WithInitialElectionID.
+
+	t.Log("Reconnect gRIBi client after switchover on new master.")
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(eID.Low, eID.High).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+
+	client.Start(ctx, t)
+	defer client.Stop(t)
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Reconnect gribi client.
+	client.StartSending(ctx, t)
+
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Log("Try to connect gRIBi client again, retrying...")
+		client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(eID.Low, eID.High).
+			WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+		client.Start(ctx, t)
+		client.StartSending(ctx, t)
+		if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+			t.Fatalf("Await got error during session negotiation for client: %v", err)
+		}
+	}
+
+	if *deviations.GRIBIDelayedAckResponse {
+		time.Sleep(4 * time.Minute)
+	}
+
+	t.Log("Validate if partially ACKed entries of IPBlock2 are present as FIB_PROGRAMMED using a get RPC.")
+
+	gr, err = args.client.Get().
+		WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithAFT(fluent.IPv4).
+		Send()
+	if err != nil {
+		t.Fatalf("Error encountered during gRIBI Get operation, got:%v", err)
+	}
+	if len(fibProgrammedEntries) != 0 {
+		for _, ipv4Prefix := range fibProgrammedEntries {
+			for _, r := range gr.GetEntry() {
+				if r.GetIpv4().Prefix == ipv4Prefix {
+					if got := r.GetFibStatus().String(); got != "PROGRAMMED" {
+						t.Fatalf("gRIBI entry %v is not programmed in FIB, got:%v, want: PROGRAMMED", r.GetIpv4().Prefix, got)
+					}
+					continue
+				}
+			}
+		}
+	}
+
+	// Check for coredumps in the DUT and validate that none are present post failover.
+	// Reconnect gnoi connection after switchover.
+	gnoiClient = dut.RawAPIs().GNOI().New(t)
```

Comment:
why do you need a new gNOI client? Can't just use the default one?

Reply:
@mojiiba requested to add this.  I will remove it now.